web/models.py

```
@@ -2818,6 +2864,41 @@ def __str__(self):
+
+    class Meta:
+        unique_together = [("user", "topic"), ("user", "reply")]
+        constraints = [
+            models.CheckConstraint(
+                check=(
+                    models.Q(topic__isnull=False, reply__isnull=True)
+                    | models.Q(topic__isnull=True, reply__isnull=False)
+                ),
+                name="vote_topic_xor_reply",
+            )
+        ]
```

Ruff (0.8.2)

Mutable class attributes should be annotated with `typing.ClassVar`

(RUF012)